app/src/main/java/com/github/se/polyfit/data/repository/MealRepository.kt

```diff
@@ -92,7 +83,7 @@ class MealRepository(
     } else {
       isDataOutdated = true
     }
-    mealDao.deleteByFirebaseID(firebaseID)
+    mealDao.deleteById(id)
```

Comment:
I just noticed this, and I know it wasn't you who implemented this (I think it was me). However, it could be a good idea to move the mealDao.deleteById to the dispatcher to avoid the issue of the main thread being blocked. (This is just something optional that can be fixed later on.)